doc/HOWTO/documenting-macros.md

```diff
@@ -0,0 +1,54 @@
+Documenting Macros
+==================
+
+The `make doc-nits` command (or the `check-docs` action) might tell
+you something like:
+
+```text
+include/openssl/bio.h: macro BIO_set_dgram_origin(3) undocumented
+include/openssl/bio.h: macro BIO_get_dgram_origin(3) undocumented
+include/openssl/bio.h: macro BIO_set_dgram_dest(3) undocumented
+include/openssl/bio.h: macro BIO_get_dgram_dest(3) undocumented
+```
+
+and you'll want to document this.
+
+So, create a new .pod file in doc/man3/FUNCTION.pod.
+
+You might be documenting several related functions in that file,
+but don't try to think about giving it a neutral name.
+Just use the name of the first function as the file name,
+so in this case I called it:
+
+```text
+doc/man3/BIO_set_dgram_origin.pod
+```
+
+if you do use an unrelated name (like BIO_dgram.pod) then you'll get
+a warning about that.
+
+Then, you'll notice that your new file is not in test/build.info.
+That file is created by running:
+
+```shell
+make generate_doc_buildinfo
+```
+
+this will update doc/build.info.
+You should git add the result as `generate_doc_buildinfo` is not run on every build.
+
+With these two changes, running `make doc-nits` locally should
+now agree with you that you have documented all your new defines,
+but it might then complain:
+
+```text
+BIO_get_dgram_dest(3) is supposedly internal
+(maybe missing from other.syms) but is documented as public
+```
+
+If it is the case that your interface is meant to be public, then you need
+to edit the file `util/other.syms` to add the names of your `#define` functions.
+This file is probably intended to be sorted alphabetically,
```

Comment:
It is sorted alphabetically on a new major release. Otherwise the new items should be put at the end.

Reply:
So the advice is to just put things at the end, even though it looks like they could be inserted alphabetically.
Why not just ask people to insert things alphabetically?

Reply:
Because it is then clear which macros were added later in the subsequent minor releases. In the end it does not matter much where you put the identifier though in case of the other.syms file, but when the same is being done for regular functions in libcrypto.num and libssl.num, it is more obvious that the newly added functions are put at the end.